src/ORM/Association.php

```diff
@@ -400,6 +401,15 @@ public function getTarget()
                 $config = ['className' => $this->_className];
             }
             $this->_targetTable = $tableLocator->get($registryAlias, $config);
+
+            $className = $this->_getClassName($registryAlias, ['className' => $this->_className]);
+            if (!is_a($this->_targetTable, $className)) {
```

Comment:
I think that the right code here is `get_class($this->_targetTable) === $className`